extensions/notebook/src/dialog/configurePython/basePage.ts

```diff
@@ -0,0 +1,33 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ConfigurePythonModel, ConfigurePythonWizard } from './configurePythonWizard';
+import { ApiWrapper } from '../../common/apiWrapper';
+
+export abstract class BasePage {
+
+	constructor(protected readonly apiWrapper: ApiWrapper,
+		protected readonly instance: ConfigurePythonWizard,
+		protected readonly wizardPage: azdata.window.WizardPage,
+		protected readonly model: ConfigurePythonModel,
+		protected readonly view: azdata.ModelView) {
+	}
+
+	/**
+	 * This method constructs all the elements of the page.
+	 */
+	public async abstract start(): Promise<boolean>;
```

Comment:
nit: But I think something like `initialize` or `create` makes more sense since it's really just creating the UI elements.